[15.0][MIG] hr_holidays_public

pegonzalezspesol opened the pull request:
*work done*
- do it installable
- tests passing
- fix pre-commit issues
- cherry-pick #36

Commit:
Update hr public holidays code to 11.

Add correct calculation of holidays in hr_public_holidays, instead of hr_holidays_compute_days.

Remove dependancy of contracts.

Add unlink at onchnage of public holiday lines.

Update code.

Fix flake.

Update code, add calculation in hours, update views.

Update flake.

Update calendar creation with no attendances, otherwise default values were set.

Remove config of show days/hours.

Update flake.

Update readme.

Rename module.

Updated holiday reports.

Update klake and pylint.

Update flaket.

Update hr_holidays_views.xml

Add readonly to show_full_days.

Update code according with comments.

Increase coverage.

update flake8.

MiquelRForgeFlow committed:
[FIX] hr_holidays_public: Review comments

pedrobaeza committed:
[FIX+IMP] hr_holidays_public: Strip hours part + complete migration

Hours part will be in module hr_holidays_hour.

pedrobaeza committed:
[IMP] hr_holidays_public: Leave only public holidays computation

Reduce to the minimum the footprint of this module, adding only public
holidays as leaves in a transparent way by other modules.

pedrobaeza committed:
[MIG] hr_holidays_compute_days: Adapt to 11.0 + minimize

This has been totally reworked for decoupling parts, not depending now
on hr_holidays_public, and making transparent its use for compatibility
with other modules.

oca-travis committed:
[UPD] Update hr_holidays_public.pot

marylla committed:
Translated using Weblate (German)

Currently translated at 52.5% (21 of 40 strings)

Translation: hr-11.0/hr-11.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-11-0/hr-11-0-hr_holidays_public/de/

alexey-pelykh committed:
[MIG] hr_holidays_public: Migration to 12.0

oca-travis committed:
[UPD] Update hr_holidays_public.pot

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 55.0% (22 of 40 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_public/es/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-12.0/hr-12.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_public/

Commit:
Translated using Weblate (French)

Currently translated at 95.0% (38 of 40 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_public/fr/

Commit:
Translated using Weblate (Spanish)

Currently translated at 92.5% (37 of 40 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_public/es/

marylla committed:
Translated using Weblate (German)

Currently translated at 92.5% (37 of 40 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_public/de/

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (40 of 40 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_public/de/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (40 of 40 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_public/es/

OCA-git-bot committed:
[UPD] README.rst

damdam-s committed:
FIX: fix leave number days when requested across years and public holidays are defined for each year

OCA-git-bot committed:
hr_holidays_public 12.0.1.0.1

Commit:
[12.0][FIX] hr_holiday_public: Fix method return for concatenation

Commit:
[FIX] hr_holiday_public: Delete holidays public

oca-travis committed:
[UPD] Update hr_holidays_public.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-12.0/hr-12.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_public/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 45.0% (18 of 40 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_public/pt_BR/

ernestotejeda committed:
[IMP] hr_holidays_public: black, isort

ernestotejeda committed:
[MIG] hr_holidays_public: Migration to 13.0

oca-travis committed:
[UPD] Update hr_holidays_public.pot

OCA-git-bot committed:
[UPD] README.rst

ivantodorovich committed:
[IMP] hr_holidays_public: Create global calendar.event

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-holidays-13.0/hr-holidays-13.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-holidays-13-0/hr-holidays-13-0-hr_holidays_public/

MiquelRForgeFlow committed:
[FIX] hr_public_holidays: add tz argument in _work_intervals

Adapt to https://github.com/odoo/odoo/commit/cc0f6e205262f73fe83b39c6ea0e29e3ae1cf355

OCA-git-bot committed:
hr_holidays_public 13.0.1.0.1

ivantodorovich committed:
[IMP] Update categ_ids only if available

OCA-git-bot committed:
pre-commit update

pedrobaeza committed:
[FIX] hr_holidays_public: Adapt to odoo/odoo#51542

Since that PR, `_leave_intervals` is no longer called on days computations, so we
intercept the batch method, but leaving the previous one for retro-compatibility.

alexey-pelykh committed:
[UPD] Brainbean Apps => CorporateHub

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update hr_holidays_public.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
hr_holidays_public 13.0.3.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-holidays-13.0/hr-holidays-13.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-holidays-13-0/hr-holidays-13-0-hr_holidays_public/

dsolanki-initos committed:
[IMP] hr_holidays_public: black, isort, prettier

dsolanki-initos committed:
[14.0][MIG] hr_holidays_public

fshah-initos committed:
Add fix #20 to 14.0

oca-travis committed:
[UPD] Update hr_holidays_public.pot

OCA-git-bot committed:
[UPD] README.rst

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 90.2% (37 of 41 strings)

Translation: hr-holidays-14.0/hr-holidays-14.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-holidays-14-0/hr-holidays-14-0-hr_holidays_public/fr_FR/

victoralmau committed:
[FIX] hr_holidays_public: Fix number_of_days in leaves when contain holiday in leave and hour start from resource calendar in 00:01

OCA-git-bot committed:
hr_holidays_public 14.0.1.0.1

victoralmau committed:
[FIX] hr_holidays_public: Prevent error in leave form without employee set yet

OCA-git-bot committed:
hr_holidays_public 14.0.1.0.2

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 97.5% (40 of 41 strings)

Translation: hr-holidays-14.0/hr-holidays-14.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-holidays-14-0/hr-holidays-14-0-hr_holidays_public/fr/

petrus-v committed:
[14.0] hr_holidays_public: add public holidays as unusual days

in order to make a grey/disabled date in the calendar view

petrus-v committed:
[14.0] hr_holidays_public: filter public holiday by the current logged user employee country

petrus-v committed:
Update hr_holidays_public/readme/DESCRIPTION.rst

Co-authored-by: Pedro M. Baeza <pedro.baeza@tecnativa.com>

petrus-v committed:
[14.0] hr_holidays_public: filter public holiday by the current logged user employee state

petrus-v committed:
fix using wrong variable name + improve coverage

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
hr_holidays_public 14.0.2.0.0

santostelmo committed:
Set exclude_public_holidays employee_id context on _compute_number_of_hours_display

OCA-git-bot committed:
hr_holidays_public 14.0.2.0.1

pedrobaeza committed:
[FIX] hr_holidays_public: Being multi-record aware in _compute_number_of_hours_display

Followup of 7d5bc92c24f6c71bab4a81ad2184fab815f58096.

It needs to be multi-record aware.

TT33269

santostelmo committed:
[Fix] hr_holiday_public: Exclude public holidays validating leaves

Steps to reproduce the problem:
- Install project_timesheet_holidays
- Create a public holiday on 2021-12-25
- Create a time-off from 2021-12-25 to 2021-12-25
- Validate time-off

Result:
Timesheet line is created on 2021-12-25

Expected result:
No timesheet line is created on 2021-12-25 since it is a public holiday.

Injecting the proper context when validating the leave avoids the problem.

pedrobaeza committed:
[FIX] hr_holidays_public: Don't copy meeting_id

If not, the date of the previous year meeting is changed for the new
year, although a new meeting is created.

It includes a test for avoiding regressions.

TT33735

pedrobaeza commented:
Please include #36 and remember to put the proper PR title for being found.

OCA-git-bot committed:
hr_holidays_public 14.0.2.0.4

pedrobaeza commented on hr_holidays_public/tests/test_holidays_calculation.py:
```diff
@@ -4,10 +4,10 @@
-class TestHolidaysComputeDaysBase(common.SavepointCase):
+class TestHolidaysComputeDaysBase(TransactionCase):
     at_install = False
     post_install = True
```
You can remove these variables that doesn't serve for anything.

pedrobaeza commented:
Please check pre-commit.

georginacicerchia reviewed (approved):
Fuctional review +1

pedrobaeza commented on hr_holidays_public/models/hr_leave.py:
```diff
@@ -8,6 +8,17 @@
 class HrLeave(models.Model):
     _inherit = "hr.leave"
 
+    def action_validate(self):
```
This shouldn't be in this commit, but cherry picking the original commit

pegonzalezspesol replied:
Squashing I wrongly merged both commits cherry-picked and my work ... will look to see if I find a way of fixing it or I will create again the PR

pedrobaeza replied:
This is still pending

pegonzalezspesol replied:
@pedrobaeza with my last PR

changes you commented are now chery-picked in commit

ca9063544ef691b9d14eab0bdb0102fc1e72ec77

my changes are in commit

c815f50c7277d0e42620ff34bafbcdd0eb980f26

pedrobaeza commented on hr_holidays_public/models/hr_holidays_public.py:
```diff
@@ -127,8 +127,8 @@ class HrHolidaysPublicLine(models.Model):
 
-    name = fields.Char("Name", required=True)
-    date = fields.Date("Date", required=True)
+    name = fields.Char(string="Name Line", required=True)
```
Why changing this?

pegonzalezspesol replied:
> Why changing this?

to prevent next pylint-odoo error

![image](https://user-images.githubusercontent.com/55989612/149090412-869637f8-5af0-4f20-b825-2963d9e62738.png)

if there is a better way of avoiding it let me know and I will implement it like that

pedrobaeza replied:
You just need to remove `string="Name"`, not to change it. That's because the string is deduced from the field name.

pegonzalezspesol replied:
> You just need to remove `string="Name"`, not to change it. That's because the string is deduced from the field name.

done!

pegonzalezspesol committed:
[15.0][MIG] hr_holidays_public

pedrobaeza commented:
Incorrect migration commit message.

olgamarcocb committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (41 of 41 strings)

Translation: hr-holidays-14.0/hr-holidays-14.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-holidays-14-0/hr-holidays-14-0-hr_holidays_public/es/

nicomacr reviewed (approved):
LGTM, pending to change the commit message

rbellanova committed:
Translated using Weblate (Italian)

Currently translated at 95.1% (39 of 41 strings)

Translation: hr-holidays-14.0/hr-holidays-14.0-hr_holidays_public
Translate-URL: https://translation.odoo-community.org/projects/hr-holidays-14-0/hr-holidays-14-0-hr_holidays_public/it/

yankinmax commented:
@pedrobaeza can you pls take a look?

AMhadhbi commented:
l10n_de_holidays is blocked by this pull request, will be good to merge it.
Thanks

pedrobaeza commented:
There's a comment unattended: https://github.com/OCA/hr-holidays/pull/38#discussion_r782570860

You can supersede fixing such problem, but respecting attribution, and I'll merge it.

pegonzalezspesol committed:
[IMP] hr_holidays_public: black, isort, prettier

pedrobaeza commented:
Thanks

/ocabot migration hr_holidays_public
/ocabot merge nobump
